honeycombio/honeycomb-opentelemetry-ruby
# feat: add BaggageSpanProcessor

## Which problem is this PR solving?

- adds BaggageSpanProcessor to the Honeycomb OpenTelemetry Ruby extras library

## Short description of the changes

- Create the actual gem library for distribution
- Wire up unit testing locally and in CI
- Add BaggageSpanProcessor, unit tests, and smoke tests for it

## How to verify that this has the expected result

- Tests pass!
- Run the example Sinatra app yourself and see that attributes names "for_the_children" appear on not-root spans

```shell
> cd smoke-tests
> docker compose up collector app-sinatra
```

... in another terminal ...

```shell
> curl localhost:4567
```
... back in the first terminal ...
Search the Collector's console output for "for_the_children" and see something like:

```
collector-1    | Span #1
collector-1    |     Trace ID       : bbafe115d386391b60d1811c73795369
collector-1    |     Parent ID      : e2d02f9bd3df85a6
collector-1    |     ID             : f2ddd822819b91ab
collector-1    |     Name           : hello
collector-1    |     Kind           : Internal
collector-1    |     Start time     : 2024-04-05 20:23:41.190116345 +0000 UTC
collector-1    |     End time       : 2024-04-05 20:23:41.190291011 +0000 UTC
collector-1    |     Status code    : Unset
collector-1    |     Status message : 
collector-1    | Attributes:
collector-1    |      -> for_the_children: Str(another important value)
```
